src/nomad.rs

```diff
@@ -0,0 +1,96 @@
+use anyhow::Result;
+use reqwest::Client;
+use serde::{Deserialize, Serialize};
+
+use crate::config::Config;
+
+#[derive(Serialize, Deserialize, Debug)]
+struct NomadService {
+    #[serde(rename = "ID")]
+    id: String,
+    #[serde(rename = "ServiceName")]
+    name: String,
+    #[serde(rename = "Tags")]
+    tags: Vec<String>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+struct NomadServiceList {
+    services: Vec<NomadService>,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct NomadDnsTag {
+    pub hostname: String,
+    pub type_: String,
+    pub ttl: Option<i32>,
+    pub value: String,
+}
+
+pub async fn fetch_and_parse_service_tags(config: &Config) -> Result<NomadDnsTag, anyhow::Error> {
+    let client = Client::new();
+    let response = client
+        .get(format!(
+            "{}:{}/v1/service/{}",
+            config.nomad_hostname, config.nomad_port, config.nomad_service_name
+        ))
+        .send()
+        .await?
+        .error_for_status()?;
+
+    let service_detail = response.json::<Vec<NomadService>>().await?;
+
+    let dns_tags: Vec<NomadDnsTag> = service_detail
+        .into_iter()
+        .flat_map(|service| {
+            service
+                .tags
+                .into_iter()
+                .filter(|tag| tag.starts_with("external-dns."))
+                .filter_map(move |tag| parse_dns_tag(&tag))
+        })
+        .collect();
+
+    if dns_tags.len() != 1 {
+        return Err(anyhow::anyhow!(
+            "Invalid number of Nomad DNS tags, expected 1 but got {}",
```

Comment:
Consider using [bail!](https://docs.rs/anyhow/latest/anyhow/macro.bail.html) for this.